geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/util/Z3FeatureIdGenerator.scala

```diff
@@ -54,12 +55,19 @@ object Z3UuidGenerator extends RandomLsbUuidGenerator {
    */
   def createUuid(sft: SimpleFeatureType, sf: SimpleFeature): UUID = {
     val time = sft.getDtgIndex.flatMap(i => Option(sf.getAttribute(i)).map(_.asInstanceOf[Date].getTime))
-      .getOrElse(System.currentTimeMillis())
-
-    val pt = sf.getAttribute(sft.getGeomIndex) match {
-      case p: Point => p
-      case g: Geometry => g.getCentroid
+        .getOrElse(System.currentTimeMillis())
+
+    val pt = {
+      val i = sft.getGeomIndex
+      if (i == -1) {
+        GeometryUtils.zeroPoint
```

Comment:
hmm, yeah maybe i should make the point random... it's kind of a fringe case though.